File: Programmers/Lv2/Lv2_올바른괄호.py
def solution(s):
    answer = True
    stack = []
    for i in s:
        if i == ')':
            if len(stack) == 0:
                return False
            else:
                if stack[-1] == '(':
                    stack.pop()
                else:
                    return False
        else:
            stack.append(i)

    if len(stack) > 0:
        return False
    else:
        return True

File: Programmers/Lv2/Lv2_올바른괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
	int l = 0;
	int r = 0;

	if (s[0] == ')')
		return false;

	for (int i = 0; i < s.length(); i++) {
		if (s[i] == '(') 
			l++;
		else
			r++;
		if (r > l)
			return false;
	}
	if (l != r)
		return false;
	else
		return true;
}